tests/test_utils.py

```diff
@@ -12,15 +37,24 @@ def read_data(response):
 def visualise_status_finished(client, p_hash):
     status = client.get("/status/" + p_hash)
     visualise_clusters_status = read_data(status)["visualiseClusters"]
-    assert len(visualise_clusters_status) > 0
-    assert all(
-        status == "finished" for status in visualise_clusters_status.values()
-    )
+    try:
+        assert len(visualise_clusters_status) > 0
+        assert all(
+            status == "finished"
+            for status in visualise_clusters_status.values()
+        )
+    except AssertionError:
+        return False
+    return True
```

Comment:
Same for this and assign_status_finished - I think with your new wait_until, it's not the job of these methods to do any asserting any more, just to return True or False for the condition each time it's called.